### PR TITLE
Session Refactoring to bring logic more near to specs

### PR DESCRIPTION
**Pull request by Apollon77**
Description for now short: see single commits ... I will update here b tomorrow and will also check if tests are missing and possible

### CODE REVIEW

File: packages/matter.js/src/MatterController.ts [hunk 1]
```diff
@@ -662,22 +665,34 @@ export class MatterController {
...
+            fabric,
+            peerNodeId,
+            peerSessionId,
+            sharedSecret,
+            salt,
+            isInitiator,
+            isResumption,
+            idleRetransmissionTimeoutMs,
+            activeRetransmissionTimeoutMs,
+        } = args;
```
Comment (lauckhart) [inline packages/matter.js/src/MatterController.ts:695]: If I'm reading correctly, looks like you could just stick ...args into createSecureSession and avoid the de-/re-structuring?

File: packages/matter.js/src/protocol/ExchangeManager.ts [hunk 2]
```diff
@@ -146,35 +146,95 @@ export class ExchangeManager<ContextT> {
...
+                );
+                this.exchanges.set(exchangeIndex, exchange);
+                await exchange.send(MessageType.StandaloneAck, new ByteArray(0));
+                await exchange.close();
+                logger.debug(
+                    `Ignoring unsolicited message ${messageId} for protocol ${message.payloadHeader.protocolId}.`,
+                );
+            } else {
+                if (protocolHandler === undefined)
+                    throw new MatterFlowError(`Unsupported protocol ${message.payloadHeader.protocolId}`);
```
Comment (lauckhart) [inline packages/matter.js/src/protocol/ExchangeManager.ts:232]: This is coming from the wire?  Maybe log a different message from below but as warning?  Don't like malformed wire messages to create what's effectively an internal error

**Review by lauckhart — APPROVED** [review]
I didn't compare super closely to the spec but it's got to be an improvement

File: packages/matter.js/src/MatterController.ts (hunk 1 above)
Comment (Apollon77) [reply packages/matter.js/src/MatterController.ts:695]: right, adjusted in two places

File: packages/matter.js/src/protocol/ExchangeManager.ts (hunk 2 above)
Comment (Apollon77) [reply packages/matter.js/src/protocol/ExchangeManager.ts:232]: Yes and nmo ... ideally we should jhave all relevant ProtocolHandlers registered ... so when a mesage comes in with unexpected one could also be on us ...

Comment (Apollon77) [reply packages/matter.js/src/protocol/ExchangeManager.ts:232]: but I adjust to warning logging

**Pull request merged**